### PR TITLE
GPS Scouters - A new survival tool for lavaland

### PR DESCRIPTION
**Pull request by Super3222**
**This PR RELIES on https://github.com/yogstation13/yogstation/pull/239 getting merged.**

GPS scouters are a different brand of GPS's. Instead of telling you the coordinates of every single GPS (in the channel of that GPS), it will allow you to send out a signal towards GPS's to a certain extent. These signals will bounce back to your scouter, and display how far away that signal is, the tag of that gps, and in what cardinal direction it's from.

If you don't wish to pick up a GPS signal, for instance the guy you are mining with, ask him to hand you his GPS and hit your scouter against it. Thus, it will add it to the scouters "buddy list" which will ignore that GPS completely. The buddy list can be cleared by holding down ctrl and clicking on the scouter.

There are two versions. The first one, is a weaker version which maximum range is pretty decent, but the amount of GPS's it can scan is under 5 otherwise it'll go on a hefty cooldown, regardless it's cooldown multiplier is pretty long altogether.

It may sound pretty bad, but the small range it has may come to an advantage. For when miners only want to have the GPS cover a certain range. This will also better suit miners who don't want their entire chat spammed with so many GPS's, which is what the upgraded version carries.

Of course, the other version has more ranges it can cover (from low/med/long to low/med/long/longer/faraway), a shorter cd multiplier, and a much larger scanning limit (from 5 -> 20!). You may want to save up your mining points for this one, however the scanning range may be unattractive to some miners. They may prefer less spam in their chat, and want to focus more on what's closest to them instead of far away. It depends on how you want to survive and what you're looking for most likely.

The normal version will only be 400 points, but the upgraded version will be 800.
I'm debating on including the normal version with the choices you get to pick from vouchers.

In the future I'm planning on making more tweaks and adjustments, probably by buffing the upgraded version, and as always feel free to leave feedback.

I'm not a spriter... don't judge me.

Info you don't care about: I at first was going to make a syndicate tool which had a similar ability like these, but instead I ended up completing this first. It'll be exclusive to nuclear operatives is all I can say for now.

:cl: Super3222
rscadd: Introducing a new mining tool to help you survive in Lavaland! ... GPS Scouters!
rscadd: These handy devices will send a 360 signal around your location attempting to pick up on GPS's. Once they do it will send information back to the scouter stating how far away that GPS is, in which cardinal direction it is from your location, and it's GPS tag.
rscadd: These will be able to pick up on tendrils and megafuana's as well!
rscadd: There are currently two versions! First, the normal version which you can buy for 400 mining  points and comes in a voucher bundle with resonators. And then the advanced version which is much more powerful then the normal one, and costs 800 mining points
/:cl:

### CODE REVIEW

**Comment by ShadowDeath6** [comment]
coder sprites best sprites

good PR

File: code/modules/telesci/gps.dm [hunk 1]
```diff
@@ -91,24 +93,164 @@ var/list/GPS_list = list()
...
+		cooldown = TRUE
+		spawn(estimated_break)
+			cooldown = FALSE
+			playsound(loc, 'sound/machines/twobeep.ogg', 50, 1)
+			visible_message("<span class='notice'>[src] is online again!</span>")
+
+/obj/item/device/gps/scouter/proc/run_scanner_report(obj/item/device/gps/G)
+	var/turf/T = get_turf(src)
+	var/turf/GT = get_turf(G)
+	if(GT in orange(shortrange,T))
```
Comment (X-TheDark) [inline code/modules/telesci/gps.dm]: orange(6...), I guess I can deal with this...but still, Isn't it easier to approximate the range?
Try something along the lines of `T.distance(GT) <= shortrange` as per https://github.com/yogstation13/yogstation/blob/master/code/game/turfs/turf.dm#L235 (**preferable solution**)
Or `RANGE_TURFS(6, T)`, as per https://github.com/yogstation13/yogstation/blob/master/code/__HELPERS/game.dm#L2 (if you can't get the previous one to work)

File: code/modules/telesci/gps.dm [hunk 2]
```diff
@@ -91,24 +93,164 @@ var/list/GPS_list = list()
...
+		return "GPS detected within short range! Identified as [G.gpstag]. Signal is [dir2text(get_dir(get_turf(src), get_turf(G)))] from your location."
+
+	if(GT in orange(midrange,T))
```
Comment (X-TheDark) [inline code/modules/telesci/gps.dm]: orange(12...), errmmm...yeah, getting into **that** territory.

File: code/modules/telesci/gps.dm [hunk 3]
```diff
@@ -91,24 +93,164 @@ var/list/GPS_list = list()
...
+		return "GPS detected within medium range! Identified as [G.gpstag]. Signal is [dir2text(get_dir(get_turf(src), get_turf(G)))] from your location."
+
+	if(GT in orange(longrange,T))
```
Comment (X-TheDark) [inline code/modules/telesci/gps.dm]: orange(18...)...Yeah...no. This is to be changed, not just a suggestion this time, this is WAY too damn expensive for what it does.

**Comment by X-TheDark** [comment]
But what does this scouter say about the power level?

**Comment by TheSadFox** [comment]
Might add it to one of the existing voucher bundles that are mostly unused to make them more attractive. Currently everyone is pretty much picking the webbing +  capsule combo.

**Comment by Super3222** [comment]
In the latest commits:
- Rewrites calls made in run_scanner_report()
- Buddy System now works fully functional, clears out some bugs as well.
- Miners have a new option when redeeming their vouchers ... a classic GPS scouter! However I'm considering what Seiko said and may add it with another existing bundle

File: code/modules/mining/machine_vending.dm [hunk 4]
```diff
@@ -155,15 +157,16 @@
...
-	var/selection = input(redeemer, "Pick your equipment", "Mining Voucher Redemption") as null|anything in list("Survival Capsule and Explorer's Webbing", "Resonator", "Mining Drone", "Advanced Scanner")
+	var/selection = input(redeemer, "Pick your equipment", "Mining Voucher Redemption") as null|anything in list("Survival Capsule and Explorer's Webbing", "Resonator", "Mining Drone", "Advanced Scanner", "GPS Scouter")
 	if(!selection || !Adjacent(redeemer) || qdeleted(voucher) || voucher.loc != redeemer)
 		return
 	switch(selection)
 		if("Survival Capsule and Explorer's Webbing")
 			new /obj/item/weapon/storage/belt/mining(src.loc)
 			new /obj/item/weapon/survivalcapsule(src.loc)
-		if("Resonator")
+		if("Resonator + GPS Scouter")
```
Comment (oisin100) [inline code/modules/mining/machine_vending.dm:167]: Uh, You never made the "Resonator + GPS Scouter" option.

**Comment by Super3222** [comment]
I sorta lost this branch.

Either way I'm going to remake this into the actual GPS's so no one has to buy them.
And make the advanced version simply an "advanced gps" with possibly 1% chance of more features.

**Pull request closed without merging**